model/common/src/icon4py/model/common/grid/base.py

```diff
@@ -78,180 +76,220 @@ def num_cells(self):
+        Callable[
+            [gtx.FieldOffset, tuple[int, int], gtx_allocators.FieldBufferAllocatorProtocol | None],
+            gtx_common.NeighborTable,
+        ]
+    ] = _default_1d_sparse_connectivity_constructor
+
+    def __post_init__(
+        self,
+        allocator: gtx_allocators.FieldBufferAllocatorFactoryProtocol | None,
+        sparse_1d_connectivity_constructor: Callable[
+            [gtx.FieldOffset, tuple[int, int], gtx_allocators.FieldBufferAllocatorProtocol | None],
+            gtx_common.NeighborTable,
+        ],
+    ):
+        # TODO(havogt): replace `Koff[k]` by `KDim + k` syntax and remove the following line.
+        self.connectivities[dims.Koff.value] = dims.KDim
+        # 1d sparse connectivities
+        self.connectivities[dims.C2CE.value] = sparse_1d_connectivity_constructor(
+            dims.C2CE, self.get_connectivity(dims.C2E).shape, allocator=allocator
+        )
+        self.connectivities[dims.E2ECV.value] = sparse_1d_connectivity_constructor(
+            dims.E2ECV, self.get_connectivity(dims.E2C2V).shape, allocator=allocator
+        )
+        self.connectivities[dims.E2EC.value] = sparse_1d_connectivity_constructor(
+            dims.E2EC, self.get_connectivity(dims.E2C).shape, allocator=allocator
+        )
+        self.connectivities[dims.C2CEC.value] = sparse_1d_connectivity_constructor(
+            dims.C2CEC, self.get_connectivity(dims.C2E2C).shape, allocator=allocator
+        )
+        if dims.C2E2C2E2C.value in self.connectivities:  # TODO is this optional?
+            self.connectivities[dims.C2CECEC.value] = sparse_1d_connectivity_constructor(
+                dims.C2CECEC, self.get_connectivity(dims.C2E2C2E2C).shape, allocator=allocator
+            )
+        self._validate()
+
+    def _validate(self):
+        expected_connectivities = {
+            dims.C2E,
+            dims.C2V,
+            dims.E2C,
+            dims.E2C2E,
+            dims.C2E2C,
+            dims.C2E2CO,
+            dims.E2C2EO,
+            dims.V2E,
+            dims.E2V,
+            dims.E2C2V,
+            dims.V2C,
+        }
+        if not all(offset.value in self.connectivities for offset in expected_connectivities):
+            missing = {c.value for c in expected_connectivities} - self.connectivities.keys()
+            raise MissingConnectivity(f"Missing connectivities in {self.id}: {', '.join(missing)}")
+
+    @functools.cached_property
+    def size(self) -> Dict[gtx.Dimension, int]:
+        sizes = {
+            dims.KDim: self.config.num_levels,
+            dims.CellDim: self.config.num_cells,
+            dims.EdgeDim: self.config.num_edges,
+            dims.VertexDim: self.config.num_vertices,
+            # 1d sparse sizes cannot be deduced from their connectivity
+            dims.ECVDim: _1d_size(self.get_connectivity(dims.E2C2V)),
+            dims.CEDim: _1d_size(self.get_connectivity(dims.C2E)),
+            dims.ECDim: _1d_size(self.get_connectivity(dims.E2C)),
```

Comment:
or maybe we deduce it from `max(self.get_connectivity(dims.E2ECV))` (or whatever the name of the FieldOffset is)

Reply:
I might do this in a follow-up PR